homeassistant/components/flux_led/config_flow.py

```diff
@@ -0,0 +1,234 @@
+                    if not devices.get(device_id, False):
+                        devices[device_id] = {
+                            CONF_NAME: bulb["ipaddr"],
+                            CONF_HOST: bulb["ipaddr"],
+                        }
+
+            return self.async_create_entry(
+                title="FluxLED/MagicHome",
+                data={
+                    CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                    CONF_EFFECT_SPEED: DEFAULT_EFFECT_SPEED,
+                    CONF_DEVICES: devices,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {vol.Required(CONF_AUTOMATIC_ADD, default=True): bool}
+            ),
+            errors=errors,
+        )
+
+
+class OptionsFlow(config_entries.OptionsFlow):
+    """Handle flux_led options."""
+
+    def __init__(self, config_entry: config_entries.ConfigEntry) -> None:
+        """Initialize the flux_led options flow."""
+
+        self._config_entry = config_entry
+        self._global_options = None
+        self._configure_device = None
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        return await self.async_step_prompt_options()
+
+    async def async_step_prompt_options(self, user_input=None):
+        """Manage the options."""
+
+        errors = {}
+
+        if user_input is not None:
+            self._global_options = {
+                CONF_AUTOMATIC_ADD: user_input[CONF_AUTOMATIC_ADD],
+                CONF_EFFECT_SPEED: user_input[CONF_EFFECT_SPEED],
+            }
+
+            if CONF_CONFIGURE_DEVICE in user_input:
+                self._configure_device = user_input[CONF_CONFIGURE_DEVICE]
+                return await self.async_step_configure_device()
+
+            if CONF_REMOVE_DEVICE in user_input:
+                device_id = user_input[CONF_REMOVE_DEVICE]
+                config_data = copy.deepcopy(dict(self._config_entry.data))
+                del config_data[CONF_DEVICES][device_id]
+
+                self.hass.config_entries.async_update_entry(
+                    self._config_entry, data=config_data
+                )
+
+                async_dispatcher_send(
+                    self.hass, SIGNAL_REMOVE_DEVICE, {"device_id": device_id}
+                )
+
+                options_data = self._config_entry.options.copy()
+                if device_id in options_data:
+                    del options_data[device_id]
+                options_data["global"] = self._global_options
+
+                return self.async_create_entry(title="", data=options_data)
+
+            if CONF_HOST in user_input:
+                device_name = (
+                    user_input[CONF_NAME]
+                    if CONF_NAME in user_input
+                    else user_input[CONF_HOST]
+                )
+                device_id = user_input[CONF_HOST].replace(".", "_")
```

Comment:
Please check https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
I noticed this and I was trying to work through it.  It's going to require changes to the Flux_LED API to get the Mac and the Model number from the devices.  Would you have a suggestion on how to get around this for now?

Reply:
If you don't have the mac don't set a unique id.  You can always add it later

Reply:
We have the Mac but only when using bulb_scanner.  My plan was to use the Model+MAC for the unique ID but, I was going to implement it once the scanner can probe specific IP addresses instead of using a broadcast.

Right, now the bulb scanner sends a broadcast out and then listens for responses.  I plan to change it so that it can also probe a specific address when an IP is passed to it.